From cb9e651718d601c6f2ff1116b4c98c11063302f7 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 17 May 2023 16:53:24 +0100
Subject: [PATCH] ARM: dts: bcm27xx: Add stdout-path to serial0

Rather than deleting the upstream stdout-path declaration, overwrite
it with one selecting serial0, which will always be the UART mapped
to the 40-pin header (provided enable_uart=1 is specified). Doing
so has the advantage that earlycon can be configured just by adding
"earlycon" to the command line.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm270x.dtsi        | 2 +-
 arch/arm/boot/dts/bcm2711-rpi-ds.dtsi | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/bcm270x.dtsi b/arch/arm/boot/dts/bcm270x.dtsi
index bb8e7a9d1b22..c318080eb883 100644
--- a/arch/arm/boot/dts/bcm270x.dtsi
+++ b/arch/arm/boot/dts/bcm270x.dtsi
@@ -5,7 +5,7 @@ / {
 	chosen: chosen {
 		// Disable audio by default
 		bootargs = "coherent_pool=1M snd_bcm2835.enable_headphones=0";
-		/delete-property/ stdout-path;
+		stdout-path = "serial0:115200n8";
 	};
 
 	soc: soc {
diff --git a/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi b/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
index 93bd3bd2d6ff..968db6362989 100644
--- a/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
+++ b/arch/arm/boot/dts/bcm2711-rpi-ds.dtsi
@@ -34,7 +34,7 @@ arm-pmu {
 	};
 
 	chosen {
-		/delete-property/ stdout-path;
+		stdout-path = "serial0:115200n8";
 	};
 
 	aliases {
-- 
2.35.5

